CONTRIBUTING.md

```diff
@@ -29,41 +27,54 @@ the code should hopefully be easy.
 
 ### Setup
 
-You can either use [Pipenv](https://pipenv.readthedocs.io/) for this:
+We use [poetry](https://python-poetry.org) for dependency and virtual environment management. Additionally
+[pre-commit](https://pre-commit.com/) is recommended to do code formatting and checking on your code when you commit.
+Install poetry and pre-commit on your system if you have not already. Using the [pipx](https://pypa.github.io/pipx/)
```

Comment:
What is the benefit from pipx?
Why not use the [official installation guide](https://python-poetry.org/docs/#installing-with-the-official-installer) from poetry?

(asking as someone who hasn't used pipx before)

Reply:
pipx allows easy and consistent upgrading/listing/uninstalling of installed packages. It does not pollute your python environment (like pip install would). And it allows installing pre-commit and poetry the same way :)

Reply:
Reformulated the README to not put as much emphasis on pipx, also removed pre-commit as requested. Pipeline runs "poetry lock" again every time, allowing quicker noticing of issues with updated versions.